src/protocol.rs

```diff
@@ -510,6 +510,44 @@ impl<'a> Protocol<'a> {
             Wss(cow) => Wss(Cow::Owned(cow.into_owned())),
         }
     }
+
+    pub fn tag(&self) -> &'static str {
```

Comment:
Given that this duplicates information (i.e. the tag) from `impl<'a> fmt::Display for Protocol<'a>`, what do you think of reusing `tag` in `impl<'a> fmt::Display for Protocol<'a>`?

``` rust
impl<'a> fmt::Display for Protocol<'a> {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        use self::Protocol::*;
        write!(f, self, "/{}", self.tag());
        match self {
            Dccp(port) => write!(f, "/{}", port),
            Dns(s) => write!(f, "/{}", s),
            Dns4(s) => write!(f, "/{}", s),
            Dns6(s) => write!(f, "/{}", s),
            Dnsaddr(s) => write!(f, "/{}", s),
            Http => {},
```

Reply:
Oh interesting.
My first thought was to use fmt in tag (and having tag return String), but then dropped it because it felt like (potentially) unnecessary memory allocations,
But going the other way around - that's a very interesting thought.